faunadb-java-dsl/src/test/java/com/faunadb/client/dsl/DslSpec.java

```diff
@@ -892,6 +896,38 @@ public void shouldCallFunction() throws Exception {
     );
   }
 
+  static class Spell {
+    @FaunaField
+    private final String name;
+    @FaunaField
+    private final String element;
+    @FaunaField
+    private final Integer cost;
+
+    @FaunaConstructor
+    Spell(@FaunaField("name") String name, @FaunaField("element") String element, @FaunaField("cost") Integer cost) {
+      this.name = name;
+      this.element = element;
+      this.cost = cost;
+    }
+  }
+
+  @Test
+  public void shouldCreateASpellUsingEncoderDecoder() throws Exception {
+    RefV ref = query(
+      Create(Ref("classes/spells"),
+        Obj("data", encode(new Spell("Blah", "blah", 10)).get())
```

Comment:
For consistency with the rest of the query language, this should be `Obj("data", Value(new Spell(..)))`.